src/main/java/com/gmail/inverseconduit/javadoc/JavaDocAccessor.java

```diff
@@ -1,130 +1,623 @@
-            sb.append(' ');
-        }
-
-        if (deprecated)
-            sb.append("---");
-        String fullName = info.getFullName();
-        String url = info.getUrl();
-        if (url == null) {
-            sb.append("**`").append(fullName).append("`**");
-        }
-        else {
-            sb.append("[**`").append(fullName).append("`**](").append(url).append(" \"View the Javadocs\")");
-        }
-        if (deprecated)
-            sb.append("---");
-        sb.append(": ");
-
-        //get the class description
-        String description = info.getDescription();
-        int pos = description.indexOf("\n");
-        if (pos >= 0) {
-            //just display the first paragraph
-            description = description.substring(0, pos);
-        }
-        sb.append(description);
-
-        return sb.toString();
-    }
+	/**
+	 * The class modifiers to print in italics when outputting a class to the
+	 * chat.
+	 */
+	private final Set<String> classModifiersItalic;
+	{
+		ImmutableSet.Builder<String> b = new ImmutableSet.Builder<>();
+		b.add("abstract");
+		b.add("final");
+		classModifiersItalic = b.build();
+	}
+
+	/**
+	 * The class modifiers to print when outputting a class to the chat.
+	 */
+	private final Set<String> classModifiers;
+	{
+		ImmutableSet.Builder<String> b = new ImmutableSet.Builder<>();
+		b.add("annotation");
+		b.add("class");
+		b.add("enum");
+		b.add("exception");
+		b.add("interface");
+		classModifiers = b.build();
+	}
+
+	/**
+	 * The method modifiers to ignore when outputting a method to the chat.
+	 */
+	private final Set<String> methodModifiersToIgnore;
+	{
+		ImmutableSet.Builder<String> b = new ImmutableSet.Builder<>();
+		b.add("private");
+		b.add("protected");
+		b.add("public");
+		methodModifiersToIgnore = b.build();
+	}
+
+	/**
+	 * @param dir the directory to the Javadocs folder
+	 * @throws IOException if there's a problem reading a Javadoc file
+	 */
+	public JavaDocAccessor(Path dir) throws IOException {
+		dao = new JavadocDao(dir);
+	}
+
+	/**
+	 * Processes a Javadoc command.
+	 * @param chatMessage the chat message (e.g. "**javadoc: arraylist")
+	 * @param commandText the command text (e.g. "arraylist")
+	 * @return the chat response
+	 */
+	public String javadoc(ChatMessage chatMessage, String commandText) {
+		String response = generateResponse(commandText.trim());
+
+		ChatBuilder cb = new ChatBuilder();
+		cb.reply(chatMessage);
+		cb.append(response);
+		return cb.toString();
+	}
+
+	private String generateResponse(String commandTextStr) {
+		if (commandTextStr.isEmpty()) {
+			return "Type the name of a Java class (e.g. \"java.lang.String\") or a method (e.g. \"Integer#parseInt\").";
+		}
+
+		//parse the command
+		CommandTextParser commandText = new CommandTextParser(commandTextStr);
+
+		ClassInfo info;
+		try {
+			info = dao.getClassInfo(commandText.className);
+		} catch (MultipleClassesFoundException e) {
+			if (commandText.methodName == null) {
```

Comment:
um.... get all the stuff inside the catch-block into a separate method. There is no need to make the `generateResponse` longer than say... 40 LoC